specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.6/TextAnalytics.json

```diff
@@ -0,0 +1,3055 @@
+          "JPDriversLicenseNumber",
+          "JPPassportNumber",
+          "JPResidentRegistrationNumber",
+          "JPSocialInsuranceNumber",
+          "JPMyNumberCorporate",
+          "JPMyNumberPersonal",
+          "JPResidenceCardNumber",
+          "LVPersonalCode",
+          "LTPersonalCode",
+          "LUNationalIdentificationNumberNatural",
+          "LUNationalIdentificationNumberNonNatural",
+          "MYIdentityCardNumber",
+          "MTIdentityCardNumber",
+          "MTTaxIDNumber",
+          "NLCitizensServiceNumber",
+          "NLCitizensServiceNumberV2",
+          "NLTaxIdentificationNumber",
+          "NLValueAddedTaxNumber",
+          "NZBankAccountNumber",
+          "NZDriversLicenseNumber",
+          "NZInlandRevenueNumber",
+          "NZMinistryOfHealthNumber",
+          "NZSocialWelfareNumber",
+          "NOIdentityNumber",
+          "PHUnifiedMultiPurposeIDNumber",
+          "PLIdentityCard",
+          "PLNationalID",
+          "PLNationalIDV2",
+          "PLPassportNumber",
+          "PLTaxIdentificationNumber",
+          "PLREGONNumber",
+          "PTCitizenCardNumber",
+          "PTCitizenCardNumberV2",
+          "PTTaxIdentificationNumber",
+          "ROPersonalNumericalCode",
+          "RUPassportNumberDomestic",
+          "RUPassportNumberInternational",
+          "SANationalID",
+          "SGNationalRegistrationIdentityCardNumber",
+          "SKPersonalNumber",
+          "SITaxIdentificationNumber",
+          "SIUniqueMasterCitizenNumber",
+          "ZAIdentificationNumber",
+          "KRResidentRegistrationNumber",
+          "ESDNI",
+          "ESSocialSecurityNumber",
+          "ESTaxIdentificationNumber",
+          "SQLServerConnectionString",
+          "SENationalID",
+          "SENationalIDV2",
+          "SEPassportNumber",
+          "SETaxIdentificationNumber",
+          "SWIFTCode",
+          "CHSocialSecurityNumber",
+          "TWNationalID",
+          "TWPassportNumber",
+          "TWResidentCertificate",
+          "THPopulationIdentificationCode",
+          "TRNationalIdentificationNumber",
+          "UKDriversLicenseNumber",
+          "UKElectoralRollNumber",
+          "UKNationalHealthNumber",
+          "UKNationalInsuranceNumber",
+          "UKUniqueTaxpayerNumber",
+          "USUKPassportNumber",
+          "USBankAccountNumber",
+          "USDriversLicenseNumber",
+          "USIndividualTaxpayerIdentification",
+          "USSocialSecurityNumber",
+          "UAPassportNumberDomestic",
+          "UAPassportNumberInternational",
+          "Organization",
+          "Email",
+          "URL",
+          "Age",
+          "PhoneNumber",
+          "IPAddress",
+          "Date",
+          "Person",
+          "Address",
+          "All",
+          "Default"
+        ]
+      },
+      "type": "array",
+      "uniqueItems": true
+    },
+    "EntityResolutions": {
+      "properties": {
+        "dateTime.TimeRange": {
```

Comment:
I think having a `.` character in the name might cause issues during deserialization to objects, since the `.` character is usually reserved for accessing properties and methods in classes.

We may need to find a different way to represent this hierarchical nature of resolution types.  We can also take some inspiration from the Healthcare API (see HealthcareRelation) which defines large numbers of relation types using enum values to distinguish between them.  The underlying object type is the same, so these types are not used as discriminators in order to produce multiple object types during deserialization, but are instead mostly used as labels.

Reply:
@abhahn This was meant to specify a format like "{EntityName}.{ResolutionShema}". We do not have the notion of hierarchy on the resolution level. The entityName (i.e., dateTime) is already specified in the surrounding Entity object. It will redundant here anyway. So, I can  remove it. Thoughts?

Reply:
Do we agree that the underlying object has all possible resolutions defined as optional properties in it?

Reply:
In regards to your first response:  yes, I think this makes sense to just drop `EntityName`, especially if we can convey the same information in types/subtypes of the Entity itself.  This should solve the issue I was worried about with deserialization.  

For your second response:  Yes, I think it is fine for all of the resolution type arrays to be specified as optional properties, so that they won't appear when the resolutions don't apply.